code/tests/generic_322.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+Reproducing xfstest generic/322 _rename_test
+
+1. Create file foo and write some contents into it and do fsync
+2. Rename foo to bar and do fsync
+
+After a crash at random point, bar should be present and should have the same contents
+written to foo.
+
+https://github.com/kdave/xfstests/blob/master/tests/generic/322
+*/
+
+
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <string>
+#include <iostream>
+#include <dirent.h>
+#include <cstring>
+#include <errno.h>
+
+#include "BaseTestCase.h"
+#include "../user_tools/api/workload.h"
+#include "../user_tools/api/actions.h"
+#define TEST_FILE_FOO "foo"
+#define TEST_FILE_BAR "bar"
+#define TEST_MNT "/mnt/snapshot"
+#define TEST_DIR_A "test_dir_a"
+
+
+using fs_testing::tests::DataTestResult;
+using fs_testing::user_tools::api::WriteData;
+using fs_testing::user_tools::api::WriteDataMmap;
+using fs_testing::user_tools::api::Checkpoint;
+using std::string;
+
+#define TEST_FILE_PERMS  ((mode_t) (S_IRWXU | S_IRWXG | S_IRWXO))
+
+namespace fs_testing {
+namespace tests {
+
+
+class Generic322: public BaseTestCase {
+ public:
+  virtual int setup() override {
+
+    // Create test directory A.
+    int res = mkdir(TEST_MNT "/" TEST_DIR_A, 0777);
+    if (res < 0) {
+      return -1;
+    }
+
+    //Create file foo in TEST_DIR_A 
+    const int fd_foo = open(foo_path.c_str(), O_RDWR | O_CREAT, TEST_FILE_PERMS);
+    if (fd_foo < 0) {
+      return -1;
+    }
+
+    // Write some contents to the file
+    const char *buf = file_contents.c_str();
+    pwrite(fd_foo, buf, strlen(buf), 0);
+
+    sync();
+    close(fd_foo);
+    return 0;
+  }
+
+  virtual int run() override {
+
+	// Rename the foo to bar
+	if (rename(foo_path.c_str(), bar_path.c_str()) != 0) {
+		return -2;
+	}
+  
+    const int fd_bar = open(bar_path.c_str(), O_RDWR | O_CREAT, TEST_FILE_PERMS);
+    if (fd_bar < 0) {
+      return -3;
+    }
+    if (fsync(fd_bar) < 0) {
+    	return -4;
+    }
+
+    if (Checkpoint() < 0){
+      return -5;
+    }
+
+    //Close open files  
+    close(fd_bar);
+
+    return 0;
+  }
+
+  virtual int check_test(unsigned int last_checkpoint,
+      DataTestResult *test_result) override {
+
+	const int fd_bar = open(bar_path.c_str(), O_RDWR | O_CREAT, TEST_FILE_PERMS);
+	if (fd_bar < 0 && last_checkpoint >= 1) {
+        test_result->SetError(DataTestResult::kFileMetadataCorrupted);
+        test_result->error_description = " : Unable to locate bar file after crash recovery";
```

Comment:
Use error code kFileMissing. You can find all available error codes here : https://github.com/utsaslab/crashmonkey/blob/22b4e971f1ec445de7878384fe4dac7295127996/code/results/DataTestResult.h#L11